File: go/dice.go
package main

import ("fmt"
        "math/rand"
        "strconv"
)

func main() {
    r := rand.New(rand.NewSource(99))
    var dice int
    var user_input string
    for {
        fmt.Println("Let's throw dice together...")
        fmt.Println("How many faces this imaginary dice have?")
        fmt.Scan(&user_input)
        fmt.Println("Ok, so I throw a dice of", user_input, "faces and get:")
        if user_input == "exit" {
            break
        } else {
            faces, err := strconv.Atoi(user_input)
            if err != nil {
                fmt.Println("Error converting string to int!")
            } else {
                dice = r.Int()%faces + 1
                fmt.Println(dice)
            }
        }        
    }
}

File: rust/single_files/dice.rs
use std::io;
use std::time::{Instant};

fn main() {
    let start_time = Instant::now();
    println!("I am ready to throw dice for you...");
    loop {
        println!("How many faces this imaginary dice have?");
        let mut user_input = String::new();
        io::stdin()
            .read_line(&mut user_input)
            .expect("That was a strange input!");
        if user_input.trim().eq("exit"){
            println!("I am killing myself now...");
            break
        } else {
            let faces: u32 = match user_input.trim().parse(){
                Ok(n) => n,
                Err(_) => {
                    println!("Imposible to get a number from that input!");
                    continue;
                }
            };
            let result: u32 = throw_dice(start_time.elapsed().as_millis() as u32, faces);
            println!("I got a: {}", result.to_string());
        }
    }
}

fn throw_dice(time: u32, faces: u32) -> u32{
    let dice: u32 = (time%faces)+1; 
    dice
}
